pkg/controller/inferenceservice/controller.go

```diff
@@ -51,100 +50,38 @@ import (
 
 	"github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha2"
 	kfserving "github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha2"
-	knservingv1 "knative.dev/serving/pkg/apis/serving/v1"
-
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/controller"
-	"sigs.k8s.io/controller-runtime/pkg/handler"
-	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
-	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
-	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
-const (
-	ControllerName = "kfserving-controller"
-)
-
-var log = logf.Log.WithName(ControllerName)
-
-// Add creates a new InferenceService Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
-// and Start it when the Manager is Started.
-func Add(mgr manager.Manager) error {
-	return add(mgr, newReconciler(mgr))
-}
-
-// newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager) reconcile.Reconciler {
-	eventBroadcaster := record.NewBroadcaster()
-	clientSet, err := kubernetes.NewForConfig(mgr.GetConfig())
-	if err != nil {
-		log.Error(err, "unable to start manager")
-		os.Exit(1)
-	}
-
-	eventBroadcaster.StartRecordingToSink(&typedcorev1.EventSinkImpl{Interface: clientSet.CoreV1().Events("")})
-
-	mgrScheme := mgr.GetScheme()
-
-	return &ReconcileService{
-		Client: mgr.GetClient(),
-		scheme: mgrScheme,
-		Recorder: eventBroadcaster.NewRecorder(
-			mgrScheme, v1.EventSource{Component: ControllerName}),
-	}
-}
-
-// add adds a new Controller to mgr with r as the reconcile.Reconciler
-func add(mgr manager.Manager, r reconcile.Reconciler) error {
-	// Create a new controller
-	c, err := controller.New(ControllerName, mgr, controller.Options{Reconciler: r})
-	if err != nil {
-		return err
-	}
-
-	// Watch for changes to InferenceService
-	if err = c.Watch(&source.Kind{Type: &kfserving.InferenceService{}}, &handler.EnqueueRequestForObject{}); err != nil {
-		return err
-	}
-
-	kfservingController := &handler.EnqueueRequestForOwner{
-		IsController: true,
-		OwnerType:    &kfserving.InferenceService{},
-	}
-
-	// Watch for changes to Knative Service
-	if err = c.Watch(&source.Kind{Type: &knservingv1.Service{}}, kfservingController); err != nil {
-		return err
-	}
-
-	// Watch for changes to Virtual Service
-	if err = c.Watch(&source.Kind{Type: &v1alpha3.VirtualService{}}, kfservingController); err != nil {
-		return err
-	}
-
-	return nil
+// Reconciler is implemented by all subresources
+type Reconciler interface {
+	Reconcile(isvc *v1alpha2.InferenceService) error
 }
 
-var _ reconcile.Reconciler = &ReconcileService{}
+var _ reconcile.Reconciler = &InferenceServiceReconciler{}
 
 // ReconcileService reconciles a Service object
-type ReconcileService struct {
+type InferenceServiceReconciler struct {
 	client.Client
-	scheme   *runtime.Scheme
+	Log      logr.Logger
```

Comment:
What's the goal with this new logging pattern

Reply:
to void the global logger variable which can cause conflicts within package